client/app/components/onboardingGuide/GuideStatusPage.js

```diff
@@ -89,7 +90,7 @@ GuideStatusPage.propTypes = {
     }),
     PropTypes.bool.isRequired,
   ]).isRequired,
-  onboarding_data: PropTypes.objectOf(PropTypes.shape({
+  onboarding_data: PropTypes.arrayOf(PropTypes.shape({
     info_image: PropTypes.string,
     cta: PropTypes.string,
     complete: PropTypes.bool.isRequired,
```

Comment:
You should add the new key `step` to the validation as well. This should be caught in the linting by eslint-plugin-react.